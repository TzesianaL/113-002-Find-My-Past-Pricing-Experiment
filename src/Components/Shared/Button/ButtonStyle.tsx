import styled from 'styled-components'

/**
 * Styled component is used to decorate Button component.
 * Props are passed from ThemeProvider component.
 */
export const ButtonStyle = styled.button`
  color: ${(props) =>
    props.theme.button?.color || props.theme.colors?.primary || '#ffffff'};

  ${(props) =>
    props.disabled
      ? `
      background-color: ${props.theme.button?.inactiveColor || 'gray'};
      pointer: default;
    `
      : `  
    background-color: ${
      props.theme.button?.backgroundColor ||
      props.theme.colors?.background ||
      '#003D59'
    };
    cursor: pointer;
  `}

  font-size: ${(props) => props.theme.fontSize || '1'}rem;
  font-family: ${(props) => props.theme.titleFontFamily};
  width: ${(props) => props.theme.button?.width};
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  border: 0px;
  border-radius: ${(props) => props.theme.button?.borderRadius || '6px'};
  padding: 0.7em 1em;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.15);

  &:active {
    background-color: ${(props) =>
      props.theme.button?.activeBackgroundColor || 'lightgray'};
  }
`
