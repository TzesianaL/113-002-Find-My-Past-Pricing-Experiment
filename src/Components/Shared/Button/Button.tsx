import React, { FC } from 'react'
import { CSSProperties } from 'styled-components'
import { ButtonStyle } from './ButtonStyle'

export interface ButtonProps {
  /**
   * Function to handle the button click.
   * This isn't needed if the button is inactive
   */
  onClick?: () => void

  /**
   * Sets whether the button is inactive
   */
  disabled?: boolean

  style?: CSSProperties
}

/**
 * Button component creates a button with a specific text, given by the user.
 * ButtonStyle component is used to style the button
 */
const Button: FC<ButtonProps> = ({ children, onClick, disabled, style }) => {
  // const onClickHandler = disabled || !onClick ? undefined : onClick
  const handleClick = (e) => {
    if (onClick !== undefined) {
      e.preventDefault()
      onClick()
    }
  }

  return (
    <ButtonStyle
      style={style}
      onClick={(e) => {
        handleClick(e)
      }}
      disabled={disabled}
    >
      {children}
    </ButtonStyle>
  )
}

export default Button
