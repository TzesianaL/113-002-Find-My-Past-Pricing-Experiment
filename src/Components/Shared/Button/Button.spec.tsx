import React from 'react'
import 'jest-styled-components'
import * as renderer from 'react-test-renderer'
import { fireEvent, render, screen } from '@testing-library/react'
import Button from './Button'

describe('<Button />', () => {
  const handleClick = (): void => {}

  it('should render correctly', () => {
    const tree = renderer
      .create(<Button onClick={handleClick}>Button</Button>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should display Button', () => {
    const el = render(<Button onClick={handleClick}>Button</Button>)
    expect(el.getByText('Button')).toBeInTheDocument()
  })

  it('test button click', () => {
    const mockCallBack = jest.fn()
    render(<Button onClick={mockCallBack}>Button</Button>)
    fireEvent.click(screen.getByText(/button/i))
    expect(mockCallBack).toHaveBeenCalledTimes(1)
  })
})
