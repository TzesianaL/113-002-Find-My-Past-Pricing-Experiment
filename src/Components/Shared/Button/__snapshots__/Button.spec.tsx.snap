// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
.c0 {
  color: #ffffff;
  background-color: #003D59;
  cursor: pointer;
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0px;
  border-radius: 6px;
  padding: 0.7em 1em;
}

.c0:active {
  background-color: #336D89;
}

<button
  className="c0"
  onClick={[Function]}
>
  Button
</button>
`;
